src/render/filters.rs

```diff
@@ -164,6 +167,47 @@ impl ResolveFilter for CapfirstFilter {
     }
 }
 
+impl ResolveFilter for CenterFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        template: TemplateString<'t>,
+        context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let left: usize;
+        let right: usize;
+        let arg_size: usize;
+        let content = match variable {
+            Some(content) => {
+                content.render(context)?.into_owned()
+            },
+            None => return Ok("".as_content()),
+        };
+        let arg = self
+            .argument
+            .resolve(py, template, context, ResolveFailures::Raise)?
+            .expect("missing argument in context should already have raised");
+        let arg_int = arg.to_bigint();
+        match arg_int {
+            Some(arg_int) => {
+                arg_size = arg_int.to_usize().unwrap();
+                if arg_size <= content.len() {
+                    return Ok(content.into_content());
+                }
+                right = (arg_size - content.len()) / 2;
+                left = arg_size - content.len() - right;
+            },
+            None => {
+                return Ok("".as_content());
+            }
+        }
+        let string_left = " ".repeat(left);
+        let string_right = " ".repeat(right);
+        Ok((string_left + &content + &string_right).into_content())
```

Comment:
I think we can slightly optimise here by using [`with_capacity`](https://doc.rust-lang.org/std/string/struct.String.html#method.with_capacity):

```rust
       let mut centered = String::with_capacity(arg_size);
       centered.push_str(&" ".repeat(left));
       centered.push_str(&content);
       centered.push_str(&" ".repeat(right);
       Ok(centered.into_content())
```

Reply:
Done.
I didn't do the below suggestion since the result of the rust implementation is not the same as the Django implementation.
So, do we want to change the result from what is documented (and then the below implementation is probably better) or not?

Reply:
Interesting! Can you elaborate on what the difference is? We probably do want to match Django's behaviour here, but I'd like to understand better before giving a final opinion.

Reply:
When I use the below code I get the following:
assertion `left == right` failed
  left: "    django     "
 right: "     django    "

---------------------------
When left is the output from the rust code and the right is the test string literal

So, if there is odd number of spaces to insert django puts the larger number at the start of the string, rust is doing it at the end.

If my explanation is not clear ask again with what is not clear so I can try to do better